src/modules/AutoinstConfig.rb

```diff
@@ -228,6 +231,38 @@ def setProposalList(l)
       nil
     end
 
+    # Checking the environment in the installed
+    # to run a second stage if it is needed.
+    #
+    # @return [String] empty String or error messsage about missing packages.
+    def check_second_stage_environment
+      error = ""
+      if AutoinstFunctions.second_stage_required?
+        missing_packages = Profile.needed_second_stage_packages.collect do |p|
+            !Pkg.IsSelected(p)
+        end
+        unless missing_packages.empty?
+          log.warn "Second stage cannot be run due missing packages: #{missing_packages}"
+          # TRANSLATORS: %s will be replaced by a package list
+          error = format(_("AutoYaST cannot run second stage due missing packages \n%s.\n"),
+            missing_packages.join(", "))
+          unless registered?
+            if Profile.current["suse_register"] &&
+              Profile.current["suse_register"]["do_registration"] == true
+              error << _("The registration has failed.")
+              error << _("Please check your registration settings in the AutoYaST configuration file.")
```

Comment:
This will concatenate the sentences.And not leave any space in between.Like this.